utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,321 @@ impl FixedDecimal {
+                } else {
+                    0
+                }
+            };
+        } else if n <= self.magnitude {
+            self.digits
+                .truncate((crate::ops::i16_abs_sub(self.magnitude, n).add(1)) as usize);
+            self.remove_trailing_zeros_from_digits_list();
+            self.lower_magnitude = {
+                if n <= 0 {
+                    n
+                } else {
+                    0
+                }
+            };
+        } else if n <= self.upper_magnitude {
+            self.lower_magnitude = {
+                if n <= 0 {
+                    n
+                } else {
+                    0
+                }
+            };
+            self.digits.clear();
+            self.magnitude = 0;
+        } else {
+            self.digits.clear();
+            self.lower_magnitude = 0;
+            self.magnitude = 0;
+            self.upper_magnitude = 0;
+        }
+
+        #[cfg(debug_assertions)]
+        self.check_invariants();
+    }
+
+    /// Ceils the number to the power ten of n.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::FixedDecimal;
+    /// # use std::str::FromStr;
+    ///
+    /// let mut dec = FixedDecimal::from_str("3.234").unwrap();
+    /// dec.ceil(0);
+    /// assert_eq!("4", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("2.222").unwrap();
+    /// dec.ceil(-1);
+    /// assert_eq!("2.3", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("22.222").unwrap();
+    /// dec.ceil(-2);
+    /// assert_eq!("22.23", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("99.999").unwrap();
+    /// dec.ceil(-2);
+    /// assert_eq!("100.00", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("99.999").unwrap();
+    /// dec.ceil(-10);
+    /// assert_eq!("99.9990000000", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("-99.999").unwrap();
+    /// dec.ceil(-10);
+    /// assert_eq!("-99.9990000000", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("-99.999").unwrap();
+    /// dec.ceil(-2);
+    /// assert_eq!("-99.99", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("99.999").unwrap();
+    /// dec.ceil(10);
+    /// assert_eq!("10000000000", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("-99.999").unwrap();
+    /// dec.ceil(10);
+    /// assert_eq!("-00000000000", dec.to_string());
+    /// ```
+    pub fn ceil(&mut self, n: i16) {
+        let is_positive = !self.is_negative;
+        let not_zero = !self.is_zero();
+        let original_bottom_magnitude = self.nonzero_magnitude_right();
+        let original_upper_magnitude = self.upper_magnitude;
+
+        self.truncate_right(n);
+
+        if n <= original_bottom_magnitude {
+            self.lower_magnitude = n;
+        } else if n <= original_upper_magnitude && is_positive {
```

Comment:
Issue: consider the following test case and make sure it passes:

```
FixedDecimal::from_str("0.009").unwrap().ceiled(-1)
```

The result should be "0.1" -- make sure this works.

Reply:
done